src/app/plan/components/Wizard/VolumesForm.tsx

```diff
@@ -34,30 +52,119 @@ const VolumesForm = props => {
     }
   }, []);
 
+  const discoveredPersistentVolumes = (currentPlan && currentPlan.spec.persistentVolumes) || [];
+
+  useEffect(() => {
+    if (discoveredPersistentVolumes.length > 0) {
+      getPVResourcesRequest(discoveredPersistentVolumes, values.sourceCluster || '');
+      let mappedPVs;
+      if (values.persistentVolumes) {
+        mappedPVs = discoveredPersistentVolumes.map(planVolume => {
+          let pvAction = 'copy'; // Default to copy
+          if (values.persistentVolumes.length !== 0) {
+            const rowVal = values.persistentVolumes.find(v => v.name === planVolume.name);
+            if (rowVal && rowVal.selection) {
+              pvAction = rowVal.selection.action;
+            } else {
+              if (rowVal && rowVal.type) {
+                pvAction = rowVal.type;
+              }
+            }
+          }
+          // TODO do we really need to generate these meta-objects and keep them in formik?
+          // currentPlan.spec.persistentVolumes is in global state, so we could just
```

Comment:
+1 Sounds like a good idea to refactor to use currentPlan as the source of truth